Getting Started With Directus and SolidStart

atilafassina opened the pull request:
fixes #41

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+## Summary
+
+[SolidStart](https://start.solidjs.com) a meta-framework built around [SolidJS](https://solidjs.com) - SolidStart is a framework aiming to allow people and teams to built high-end applications while bringing their own opinions. In this tutorial, you will learn how to build a website using Directus as a CMS. You will store, retrieve, and use global metadata such as the site title, create new pages ...
```
```suggestion
[SolidStart](https://start.solidjs.com) a meta-framework built around [SolidJS](https://solidjs.com) - SolidStart is a framework aiming to allow people and teams to built high-end applications without being too opinionated. In this tutorial, you will learn how to build a website using Directus as a CMS. You will store, retrieve, and use global metadata such as the site title, create new pages dynamically based on Directus items, and build a blog.
```

When you say 'with their own opinions' it doesn't make as much sense as the word 'opinionated' as developers have come to know it.

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+You will need:
+
+- To install Node.js and a code editor on your computer.
+- A Directus project - follow our quickstart guide if you don't already have one.
+- SolidJS knowledge will definitely help, but will not prevent you from following this guide.
```
```suggestion
- Node.js and a code editor.
- A Directus project - [follow our quickstart guide](https://docs.directus.io/getting-started/quickstart) if you don't already have one.
- (Bonus) SolidJS knowledge will definitely help, but is not required.
```

This suggestion is mostly an act of being more direct, removing words that don't add value, and linking users to the quickstart which they may need.

phazonoverload commented on directus-solidstart/index.md:
````diff
@@ -0,0 +1,314 @@
+
+Open your terminal and run the following commands to create a new Solid project and add the Directus JavaScript SDK:
+
+```sh
+npm create solid@latest solid-directus
+```
+
+A prompt will start, here are the sequence of answers provided for this exercise:
+
+```text
+✔ Which template do you want to use?
+  › bare
+
+✔ Server Side Rendering?
+  › yes
+
+✔ Use TypeScript?
+  › no
+```
````
````suggestion
```sh
npm create solid@latest solid-directus
✔ Which template do you want to use?
  › bare
✔ Server Side Rendering?
  › yes
✔ Use TypeScript?
  › no
```
````

Combining small steps to shorten the read. Can we also pin this to a version to limit the chance this guide becomes incorrect in the future?

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+## Create a Directus Helper
+
+We now need to setup the Directus SDK. We will add our data fetchers in this same file so we can handle caching and requests all at the same place. Create a new file `directus.js` inside of the src/libs directory:
```
```suggestion
We now need to setup the Directus SDK. We will add our data fetchers in this same file so we can handle caching and requests all at the same place. Create a new file `directus.js` inside of the `src/libs` directory:
```

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+We have some unused `import` statements at the moment since now we are only instantiating the Directus client.
+
+> Remember to replace `"< your project URL >"` with your actual URL, you can find it in your Directus dashboard.
```
```suggestion
:::tip Provide Your Directus Project URL

Remember to replace `< your project URL >` with your actual URL, you can find it in your Directus dashboard.

:::
```

Formatting. Also removed quotes as we want the quotes to not be replaced.

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+> Remember to replace `"< your project URL >"` with your actual URL, you can find it in your Directus dashboard.
+
+It may be useful to create an `.env` file and store the URL there in case you use different projects for production, development, etc. Otherwise it is not a private value, access is managed through the Directus dashboard.
```
Let's remove this and assume everything is public for this post. We can mention this in __Next Steps__.

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+    return await directus.request.readItems("global");
+  } catch (e) {
+    console.error(e);
+
```
Remove extra space

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+    return null;
+  }
+}, `globals`);
```
```suggestion
}, "globals");
```

Wouldn't normally bother me but you wrap in double quotes below and immediately I couldn't match up what you wrote with the code because of the differing quote types.

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+Create a new collection called pages - make the Primary ID Field a "Manually Entered String" called slug, which will correlate with the URL for the page. For example about will later correlate to the page [localhost:3000/about](http://localhost:3000/about).
+
+Create a text input field called title and a WYSIWYG input field called content. In Roles & Permissions, give the Public role read access to the new collection. Create 3 items in the new collection - here's some sample data.
```
Note to self - leave this resolved - move sample data into a common space and link it here.

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+Create a text input field called title and a WYSIWYG input field called content. In Roles & Permissions, give the Public role read access to the new collection. Create 3 items in the new collection - here's some sample data.
+
+Now, we go to our `src/lib/directus.js` and create our new fetcher.
```
```suggestion
Open `src/lib/directus.js` and create our new fetcher.
```

More succinct and instructional

phazonoverload commented on directus-solidstart/index.md:
````diff
@@ -0,0 +1,314 @@
+  }, slug);
+```
+
+Again, this method will attempt to fetch information from a given `slug`. And if there is no information, it will return `null`.
````
```suggestion
This method attempts to fetch information from a given `slug`. If there is no information, it will return `null`.
```

More direct

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+## Dynamic Routes With Directus
+
+Create a new collection called pages - make the Primary ID Field a "Manually Entered String" called slug, which will correlate with the URL for the page. For example about will later correlate to the page [localhost:3000/about](http://localhost:3000/about).
```
```suggestion
Create a new collection called pages - make the Primary ID Field a "Manually Entered String" called slug, which will correlate with the URL for the page. For example, `about` will later correlate to the page [localhost:3000/about](http://localhost:3000/about).
```

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+Again, this method will attempt to fetch information from a given `slug`. And if there is no information, it will return `null`.
+
+Inside of `src/routes`, create a new a file called `slug.jsx`. This is a dynamic route, so a single file can be used for all of the top-level pages.
```
```suggestion
Inside of the `src/routes` directory, create a new a file called `slug.jsx`. This is a dynamic route, so a single file can be used for all of the top-level pages.
```

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+Inside of `src/routes`, create a new a file called `slug.jsx`. This is a dynamic route, so a single file can be used for all of the top-level pages.
+
+> It's important to note that exact matches take precedence over dynamic routes. If it exists, a `about.jsx` file will render instead of `[slug].jsx` for the `/about` URL.
```
```suggestion
:::tip Dynamic Route Precedence

It's important to note that exact matches take precedence over dynamic routes. If it exists, a `about.jsx` file will render instead of `[slug].jsx` for the `/about` URL.

:::
```

phazonoverload replied:
I'd also move this below the code snippet as more of a footnote to the section.

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+## Creating Blog Posts
+
+Create a new collection called authors with a single text input field called name. Create one or more authors.
```
```suggestion
Create a new collection called `authors` with a single text input field called `name`. Create one or more authors.
```

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+Create a new collection called authors with a single text input field called name. Create one or more authors.
+
+Then, create a new collection called posts - make the Primary ID Field a "Manually Entered String" called slug, which will correlate with the URL for the page. For example hello-world will later correlate to the page `localhost:3000/blog/hello-world`.
```
```suggestion
Then, create a new collection called `posts` - make the Primary ID Field a "Manually Entered String" called `slug`, which will correlate with the URL for the page. For example, `hello-world` will later correlate to the page `localhost:3000/blog/hello-world`.
```

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+In Roles & Permissions, give the Public role read access to the `authors`, `posts`, and `directus_files` collections.
+
+Create 3 items in the posts collection -[here's some sample data](https://github.com/directus/examples/blob/main/website-next13/demo-data).
```
Note to self, please don't resolve - centralize data and update link

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+## List Blog Posts
+
+Time to add another fetcher. Back into our `src/lib/directus.js`, add now a request to read multiple items from `"posts"`.
```
```suggestion
Open `src/lib/directus.js`, add a helper to request all items from the `posts` collection.
```

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+![A page with a title of "Blog". On it is a list of three items - each with a title, author, and date. The title is a link.](blog-list.webp)
+
+This time, we don't want this route to throw ever. What we want is to just show an empty list if by any chance we are not successful, so not rendering the list is enough.
```
```suggestion
This fetcher should not throw an error, and instead show an empty list if the request is not successful.
```

This is more direct about the task, and why.

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+This time, we don't want this route to throw ever. What we want is to just show an empty list if by any chance we are not successful, so not rendering the list is enough.
+
+In our `src/routes`, we can now create a `blog` directory. And within the newly create directory, an `index.jsx`.
```
```suggestion
Iside of the `src/routes` directory, now create a `blog` directory. And within this newly created directory, an `index.jsx` file:
```

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+## Add Navigation
+
+While not strictly Directus-related, there are now several pages that aren't linked to each other. Create the file `src/routes/app.jsx` to add a navigation above the main content. Don't forget to use your specific page slugs.
```
```suggestion
While not strictly Directus-related, there are now several pages that aren't linked to each other. Create a `src/routes/app.jsx` file to add a navigation above the main content. Don't forget to use your specific page slugs.
```

phazonoverload commented on directus-solidstart/index.md:
```diff
@@ -0,0 +1,314 @@
+
+## Next Steps
+
+Through this guide, you have set up a Next project, created a Directus helper, and used it to query data. You have used a singleton collection for global metadata, dynamically created pages, as well as blog listing and post pages.
```
```suggestion
Through this guide, you have set up a SolidStart project, created a Directus helper, and used it to query data. You have used a singleton collection for global metadata, dynamically created pages, as well as blog listing and post pages.
```

phazonoverload reviewed (changes requested):
Good post - all change suggestions are minor and mostly related to being more direct to the reader and removing filler words.

atilafassina commented:
> Good post - all change suggestions are minor and mostly related to being more direct to the reader and removing filler words.

Thanks a lot for the thorough review!
I believe I have addressed all your suggestions :)

phazonoverload reviewed (approved):
Thank you for writing this, and I'm happy to consider it done in it's current state. I'll schedule it in to be published. 

This repo is just for authoring/reviewing - the actual piece will be published via a Directus project, so I'll take it from here. I'll close this issue to keep the main branch clean, but this is a acceptance of your work.

Could you email me at [kevin@directus.io](mailto:kevin@directus.io) and I'll share what we need to get you paid.
